frontend/src/pages/agreements/approve/ApproveAgreement.js

```diff
@@ -76,16 +116,44 @@ const ApproveAgreement = () => {
         });
     };
 
-    const handleApprove = () => {
+    const approveStep = async () => {
+        const data = {
+            workflow_step_action: "APPROVE",
+            workflow_step_id: stepId,
+            notes: notes
+        };
+
+        await workflowApprove(data)
+            .unwrap()
+            .then((fulfilled) => {
+                console.log(`SUCCESS of workflow-approve: ${JSON.stringify(fulfilled, null, 2)}`);
+                setAlert({
+                    type: "success",
+                    heading: "Approval Saved",
+                    message: `The approval to change Budget Lines has been saved.`
+                });
+            })
+            .catch((rejected) => {
+                console.error(`ERROR with workflow-approve: ${JSON.stringify(rejected, null, 2)}`);
+                setAlert({
+                    type: "error",
+                    heading: "Error",
+                    message: "An error occurred while saving the approval.",
+                    redirectUrl: "/error"
+                });
+            });
+    };
+
+    const handleApprove = async () => {
         setShowModal(true);
         setModalProps({
             heading:
                 "Are you sure you want to approve these budget lines for Planned Status? This will subtract the amounts from the FY budget.",
             actionButtonText: "Approve",
             secondaryButtonText: "Cancel",
-            handleConfirm: () => {
-                alert("Not implemented yet");
-                navigate("/agreements");
+            handleConfirm: async () => {
+                await approveStep();
+                await navigate("/agreements");
```

Comment:
TypeScript says we can remove the `await` on 156

Reply:
hmmm, that's how it's done on ApproveAgreementEditForm's handleDraft